client/client_test.go

```diff
@@ -1686,6 +1720,188 @@ func TestPublishTargetsDelgationSuccessNeedsToDownloadRoles(t *testing.T) {
 		[]string{"targets/a/b"})
 }
 
+// Ensure that two clients can publish delegations with two different keys and
+// the changes will not clobber each other.
+func TestPublishTargetsDelgationFromTwoRepos(t *testing.T) {
+	var tempDirs [2]string
+	for i := 0; i < 2; i++ {
+		tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+		assert.NoError(t, err, "failed to create a temporary directory: %s", err)
+		defer os.RemoveAll(tempBaseDir)
+		tempDirs[i] = tempBaseDir
+	}
+
+	gun := "docker.com/notary"
+	ts := fullTestServer(t)
+	defer ts.Close()
+
+	// this happens to be the client that creates the repo, but can also
+	// write a delegation
+	repo1, _ := initializeRepo(t, data.ECDSAKey, tempDirs[0], gun, ts.URL, true)
+	defer os.RemoveAll(repo1.baseDir)
```

Comment:
No, not at all - good catch, thanks :) This change was originally part of #384, so it's an artifact of that refactor.